src/test/java/group1/mips_simulator/components/InstructionExecutionsTest.java

```diff
@@ -205,4 +206,60 @@ void execute_jz() {
         computer.executeInstruction(instruction);
         assertEquals(20, computer.cpu.regfile.getPC().read());
     }
+
+    short shifter(boolean shiftLeft, boolean shiftLogically, short value, short count) {
+        if (shiftLeft) {
+            value = (short) (value << count);
+        } else {
+            if (shiftLogically) {
+                value = (short) (value >>> count); // Signed
+            } else {
+                // arithmetic
+                value = (short) (value >> count); // Unsigned
+            }
+        }
+        return value;
+    }
+
+    @Test
+    void shitTests() {
```

Comment:
👀👀

Reply:
Lol oops. I meant "shift" tests. But the title is still accurate....